docs/contributing/core/index.md

```diff
@@ -0,0 +1,268 @@
+
+Then rerun buildout to install the source packages to edit.
+
+```shell
+./bin/buildout
+```
+
+```{seealso}
+For tips on working with `mr.developer`, see {doc}`mrdeveloper`.
+```
+
+
+## Test locally
+
+If you change the expected behavior of a feature in a package, you should write a test to cover the change.
+
+To run a test for the specific package that you modified, use the `-m` option followed by the package name, as shown in the following example.
+
+```shell
+./bin/test -m plone.app.caching
+```
+
+If any test fails, do not commit and push the changes.
+Instead write a test that passes.
+
+After the package level tests pass with your change, you can run the full unit test suite to ensure other packages aren't affected by the change.
+It takes 5-10 minutes to run the full unit test suite.
+
+```shell
+# Run unit tests
+./bin/test
+```
+
+If you run acceptance tests with the `--all` option, it will repeatedly launch and close browser windows that gain focus, disrupting you from doing any other work.
+This takes 30-40 minutes to run.
+If you can't afford this disruption, you can defer it to Jenkins.
+
+```shell
+# Run acceptance tests
+./bin/test --all
+```
+
+
+## Change log
+
+All changes require a change log entry.
+
+For packages that use [towncrier](https://pypi.org/project/towncrier/) to produce change logs, see {ref}`contributing-change-log-label`.
+
+For packages that don't use towncrier, edit either {file}`CHANGES.rst`, {file}`CHANGES.txt`, or {file}`HISTORY.txt` in each package you have modified, adding a summary of the change.
+New change log entries should be added at the top of {file}`CHANGES.rst`.
+
+
+## Update `checkouts.cfg`
+
+If you didn't do so earlier, edit the file {file}`checkouts.cfg` at the root, and add your changed packages only to the `auto-checkout` list.
+Remove any packages that you previously added to this file, but did not change.
+Leave any pre-existing packages in the original list.
+
+The Plone release manager will check this file to see which packages you have updated to make a proper release.
+
+```{seealso}
+{doc}`release`
+```
+
+
+## Create a pull request
+
+After you have completed all the foregoing steps, push your changes to a remote branch and create a pull request in GitHub.
+
+If you are working from an issue, return to the original issue, and add a link to the pull request.
+
+
+## Jenkins and mr.roboto
+
+Plone has a continuous integration ({term}`CI`) setup and follows CI rules.
+
+When you push a change to a Plone package, there may be GitHub workflows that run automatically.
+The CI package [mr.roboto](https://github.com/plone/mr.roboto) will perform some checks and suggest that you run Jenkins after all other CI runs.
+
+For each Plone and Python version, Jenkins runs two jobs: one for the package itself (which will give you a fast feedback, within 10 minutes) and one on the full `coredev` build (which can take up to an hour, but makes sure no other packages are affected by your change).
```

Comment:
These days Jenkins does not run a job for one specific package. For most packages there are some github actions workflows that do that, as well as some lint checks.

There are usually several jobs that run all Plone tests on Jenkins -- basically one for each combination of Plone version + Python version where that branch of the package is used, but some of them may skipped to reduce resource use (i.e. for Plone 6 we run this job on Python 3.8 and 3.11 but not 3.9 or 3.10)

Reply:
I'm actually OK with purging this detail and not trying to document it here. It doesn't belong in a how-to guide, but more in explanation or reference.

Done in 6ef5c58872b6b68c76a8563c41a38b506ded8496